File: src/test/java/com/isycat/YamllintMavenPluginTest.java
package com.isycat;


import com.isycat.yamllintplugin.YamlLintException;
import org.apache.maven.plugin.testing.MojoRule;
import org.junit.Rule;
import org.junit.Test;

import java.io.File;

import static org.junit.Assert.*;

public class YamllintMavenPluginTest {
    @Rule
    public MojoRule rule = new MojoRule() {
        @Override
        protected void before() throws Throwable {
        }

        @Override
        protected void after() {
        }
    };

    @Test
    public void testFailureProject() throws Exception {
        File pom = new File("target/test-classes/line-too-long-project/");
        assertNotNull(pom);
        assertTrue(pom.exists());

        YamllintMavenPlugin yamllintMavenPlugin = (YamllintMavenPlugin) rule.lookupConfiguredMojo(pom, "lint");

        boolean thrown = false;
        try {
            yamllintMavenPlugin.execute();
        } catch (final YamlLintException e) {
            thrown = true;
            assertEquals("1 error(s) found while parsing yaml files.", e.getMessage());
        }

        assertEquals("Expected YamlLintException but no error was thrown", true, thrown);
    }

    @Test
    public void testNoErrorProject() throws Exception {
        File pom = new File("target/test-classes/successful-project/");
        assertNotNull(pom);
        assertTrue(pom.exists());

        YamllintMavenPlugin yamllintMavenPlugin = (YamllintMavenPlugin) rule.lookupConfiguredMojo(pom, "lint");
        yamllintMavenPlugin.execute();
    }
}

File: yamllint-maven-plugin/src/test/kotlin/com/isycat/yamllintplugin/YamllintMavenPluginTest.kt
package com.isycat.yamllintplugin

import org.junit.Test
import java.io.File
import org.junit.Assert.assertTrue

class YamllintMavenPluginTest {

    @Test
    @Throws(Exception::class)
    fun testFailureProject() {
        val sourceDir = File("src/test/resources/line-too-long-project/")
        val configFile = File(sourceDir, ".yamllint")

        val yamllintMavenPlugin = YamllintMavenPlugin()
        yamllintMavenPlugin.sourceDir = sourceDir
        yamllintMavenPlugin.configFile = configFile

        var thrown = false
        try {
            yamllintMavenPlugin.execute()
        } catch (e: Exception) {
            // Check if it's our exception or wrapped in MojoExecutionException
            val cause = e.cause
            if (e.javaClass.name.endsWith("YamlLintException") ||
                (cause != null && cause.javaClass.name.endsWith("YamlLintException"))
            ) {
                thrown = true
                val msg = cause?.message ?: e.message
                assertTrue(msg!!.contains("error(s) found while parsing yaml files."))
            } else {
                throw e
            }
        }

        assertTrue("Expected YamlLintException but no error was thrown", thrown)
    }

    @Test
    @Throws(Exception::class)
    fun testNoErrorProject() {
        val sourceDir = File("src/test/resources/successful-project-clean/")

        val yamllintMavenPlugin = YamllintMavenPlugin()
        yamllintMavenPlugin.sourceDir = sourceDir
        yamllintMavenPlugin.configFile = File(sourceDir, ".yamllint")

        yamllintMavenPlugin.execute()
    }
}
